.gitlab-ci.yml

```diff
@@ -192,14 +194,56 @@ build-windows:
 
 publish-docker:
   stage:                           publish
-  only:                            *releaseable_branches
-  cache:                           {}
   dependencies:
     - build-linux
   tags:
-    - shell
+    - kubernetes-parity-build
+  environment:
+    name: parity-build
+  cache:                           {}
+  only:                            *releaseable_branches
+  except:
+    - master
+  image:                           docker:stable
+  services:
+    - docker:dind
+  variables:
+    DOCKER_HOST:                   tcp://localhost:2375
+    DOCKER_DRIVER:                 overlay2
+    GIT_STRATEGY:                  none
+    # DOCKERFILE:                  artifacts/Dockerfile
+    CONTAINER_IMAGE:               parity/parity
+  before_script:
+    - test "$Docker_Hub_User_Parity" -a "$Docker_Hub_Pass_Parity"
+        || ( echo "no docker credentials provided"; exit 1 )
+    - docker login -u "$Docker_Hub_User_Parity" -p "$Docker_Hub_Pass_Parity"
+    - docker info
   script:
-    - scripts/gitlab/publish-docker.sh parity
+    # we stopped pushing nightlies to dockerhub, will push to own registry prb.
+    - case "${SCHEDULE_TAG:-${CI_COMMIT_REF_NAME}}" in
+        $SCHEDULE_TAG)
+          echo "Docker TAG - ${CONTAINER_IMAGE}:${SCHEDULE_TAG}";;
+        beta)
+          echo "Docker TAGs - ${CONTAINER_IMAGE}:beta, ${CONTAINER_IMAGE}:latest";
+          docker build --no-cache
+              --build-arg VCS_REF="${CI_COMMIT_SHORT_SHA}"
+              --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')"
+              --tag ${CONTAINER_IMAGE}:beta
+              --tag ${CONTAINER_IMAGE}:latest
+              --file artifacts/Dockerfile .;
+          docker push ${CONTAINER_IMAGE}:beta;
+          docker push ${CONTAINER_IMAGE}:latest ;;
+        stable|*)
+          echo "Docker TAGs - ${CONTAINER_IMAGE}:${CI_COMMIT_REF_NAME}";
+          docker build --no-cache
+              --build-arg VCS_REF="${CI_COMMIT_SHORT_SHA}"
+              --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')"
+              --tag ${CONTAINER_IMAGE}:$CI_COMMIT_REF_NAME
+              --file artifacts/Dockerfile .;
+          docker push ${CONTAINER_IMAGE}:$CI_COMMIT_REF_NAME ;;
```

Comment:
maybe it makes sense to test the image before the push?

Reply:
I would suggest to put a test into the Dockerfile as in the substrate dockerfile.

Reply:
good point